### PR TITLE
Users shouldn't receive a FORBIDDEN (403) response code from DatawavePrincipalLoginModule for datawave problems

### PR DESCRIPTION
**Pull request by billoley**
Users shouldn't receive a FORBIDDEN (403) response code from DatawavePrincipalLoginModule for datawave problems

See https://github.com/NationalSecurityAgency/datawave/issues/2546

### CODE REVIEW

File: web-services/security/src/main/java/datawave/security/login/DatawavePrincipalLoginModule.java [hunk 1]
```diff
@@ -412,94 +426,114 @@ protected DatawaveCredential getDatawaveCredential() throws LoginException {
...
+                    // should result in a SERVICE_UNAVAILABLE (503) response code in DatawaveAuthenticationMechanism.sendChallenge
+                    throw new LoginException("Unable to authenticate: " + e.getMessage());
                 }
             } else {
                 try {
                     identity = new DatawavePrincipal(jwtTokenHandler.createUsersFromToken(credential.getJwtToken()));
                 } catch (Exception e) {
                     log.debug("Failing login due to JWT token exception " + e.getMessage(), e);
-                    throw new FailedLoginException("Unable to authenticate: " + e.getMessage());
+                    // should result in an UNAUTHORIZED (501) response code in DatawaveAuthenticationMechanism.sendChallenge
```
Comment (keith-ratcliffe) [inline web-services/security/src/main/java/datawave/security/login/DatawavePrincipalLoginModule.java]: I think comment should say 401

File: web-services/security/src/main/java/datawave/security/login/DatawavePrincipalLoginModule.java [hunk 2]
```diff
@@ -412,94 +426,114 @@ protected DatawaveCredential getDatawaveCredential() throws LoginException {
...
+                    throw new CredentialException("Unable to authenticate: " + e.getMessage());
                 }
             }
         }
         if (getUseFirstPass()) {
             sharedState.put("javax.security.auth.login.name", alias);
             sharedState.put("javax.security.auth.login.password", credential.getCertificate());
         }
-        if (trace)
+        if (log.isTraceEnabled()) {
```
Comment (keith-ratcliffe) [inline web-services/security/src/main/java/datawave/security/login/DatawavePrincipalLoginModule.java]: I could be missing something, but unclear why the switch from `if (trace)` to `if (log.isTraceEnabled())`

**Comment by keith-ratcliffe** [comment]
These changes LGTM, nits aside

File: web-services/security/src/main/java/datawave/security/login/DatawavePrincipalLoginModule.java (hunk 1 above)
Comment (billoley) [reply web-services/security/src/main/java/datawave/security/login/DatawavePrincipalLoginModule.java]: fixed

File: web-services/security/src/main/java/datawave/security/login/DatawavePrincipalLoginModule.java (hunk 2 above)
Comment (billoley) [reply web-services/security/src/main/java/datawave/security/login/DatawavePrincipalLoginModule.java]: Was changing all instanced to isTraceEnabled, then reconsidered and apparently forgot some
Fixed

**Comment by ivakegg** [comment]
Please take a look at the microservices as I expect this will require changes there as well.  However this looks good to me.

**Review by ivakegg — APPROVED** [review]

**Review by keith-ratcliffe — APPROVED** [review]

**Pull request merged**